crates/stages/src/stages/headers.rs

```diff
@@ -74,26 +74,18 @@ impl<DB: Database, D: HeaderDownloader, C: Consensus, H: HeadersClient, S: Statu
         let stage_progress = input.stage_progress.unwrap_or_default();
         self.update_head::<DB>(db, stage_progress).await?;
 
+        // Lookup the head and tip of the sync range
+        let (head_hash, tip) = self.get_head_and_tip(db, stage_progress).await?;
+
         // Lookup the last stored header
-        let last_hash = db.get_block_hash(stage_progress)?;
         let last_header =
-            db.get::<tables::Headers>((stage_progress, last_hash).into())?.ok_or({
-                DatabaseIntegrityError::Header { number: stage_progress, hash: last_hash }
+            db.get::<tables::Headers>((stage_progress, head_hash).into())?.ok_or({
+                DatabaseIntegrityError::Header { number: stage_progress, hash: head_hash }
             })?;
-        let head = SealedHeader::new(last_header, last_hash);
-
-        let forkchoice = self.next_fork_choice_state(&head.hash()).await;
-        if let Some(number) = db.get::<tables::HeaderNumbers>(forkchoice.head_block_hash)? {
-            if number < head.number {
-                // Nothing to do here
-                warn!("Consensus reported old head {}", forkchoice.head_block_hash);
-                return Ok(ExecOutput { stage_progress, done: true, reached_tip: true })
-            }
-        }
```

Comment:
Why is this removed? @onbjerg said that this may require a rewind if it actually happened? Might be wrong.